builtin/providers/azurerm/resource_arm_traffic_manager_endpoint.go

```diff
@@ -0,0 +1,272 @@
+	name := id.Path[path.Base(endpointType)]
+
+	resp, err := client.Get(resGroup, profileName, path.Base(endpointType), name)
+	if resp.StatusCode == http.StatusNotFound {
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Error making Read request on TrafficManager Endpoint %s: %s", name, err)
+	}
+	endpoint := *resp.Properties
+
+	d.Set("name", resp.Name)
+	d.Set("endpoint_status", *endpoint.EndpointStatus)
+
+	if endpoint.TargetResourceID != nil {
+		d.Set("target_resource_id", *endpoint.TargetResourceID)
+	}
+
+	if endpoint.Target != nil {
+		d.Set("target", *endpoint.Target)
+	}
+
+	if endpoint.Weight != nil {
+		d.Set("weight", int(*endpoint.Weight))
+	}
+
+	if endpoint.Priority != nil {
+		d.Set("priority", int(*endpoint.Priority))
+	}
+
+	if endpoint.EndpointLocation != nil {
+		d.Set("endpoint_location", *endpoint.EndpointLocation)
+	}
+
+	if endpoint.EndpointMonitorStatus != nil {
+		d.Set("endpoint_monitor_status", *endpoint.EndpointMonitorStatus)
+	}
+
+	if endpoint.MinChildEndpoints != nil {
+		d.Set("min_child_endpoints", *endpoint.MinChildEndpoints)
+	}
+
+	return nil
+}
+
+func resourceArmTrafficManagerEndpointDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).trafficManagerEndpointsClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	endpointType := d.Get("type").(string)
+	profileName := id.Path["trafficManagerProfiles"]
+
+	// endpoint name is keyed by endpoint type in ARM ID
+	name := id.Path[path.Base(endpointType)]
+
+	_, err = client.Delete(resGroup, profileName, path.Base(endpointType), name)
+
+	return err
+}
+
+func getArmTrafficManagerEndpointProperties(d *schema.ResourceData) *trafficmanager.EndpointProperties {
+	var endpointProps trafficmanager.EndpointProperties
+
+	if targetResID := d.Get("target_resource_id").(string); targetResID != "" {
+		endpointProps.TargetResourceID = &targetResID
+	}
+
+	if target := d.Get("target").(string); target != "" {
+		endpointProps.Target = &target
+	}
+
+	if status := d.Get("endpoint_status").(string); status != "" {
+		endpointProps.EndpointStatus = &status
+	}
+
+	if weight := d.Get("weight").(int); weight != 0 {
+		w64 := int64(weight)
+		endpointProps.Weight = &w64
+	}
+
+	if priority := d.Get("priority").(int); priority != 0 {
+		p64 := int64(priority)
+		endpointProps.Priority = &p64
+	}
+
+	if location := d.Get("endpoint_location").(string); location != "" {
+		endpointProps.EndpointLocation = &location
+	}
+
+	if minChildEndpoints := d.Get("min_child_endpoints").(int); minChildEndpoints != 0 {
+		mci64 := int64(minChildEndpoints)
+		endpointProps.MinChildEndpoints = &mci64
+	}
+
+	return &endpointProps
+}
+
+func validateAzureRMTrafficManagerEndpointType(i interface{}, k string) (s []string, errors []error) {
+	valid := map[string]struct{}{
+		"Microsoft.Network/TrafficManagerProfiles/azureEndpoints":    struct{}{},
```

Comment:
Do we really need these entire strings? Doesn't the API add these for us if we pass in `azureEndpoints`, `externalEndpoints` or `nestedEndpoints`?

We haven't had to pass type before so wondering what the path of least resistance is here